prow/prstatus/prstatus.go

```diff
@@ -222,60 +221,63 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler pullRequestQueryHandler, c
 					return
 				}
 			}
-		}
 
-		if user != nil {
-			login := user.Login
-			data.Login = true
-			// Saves login. We save the login under 2 cookies. One for the use of client to render the
-			// data and one encoded for server to verify the identity of the authenticated user.
-			http.SetCookie(w, &http.Cookie{
-				Name:    loginSession,
-				Value:   login,
-				Path:    "/",
-				Expires: time.Now().Add(time.Hour * 24 * 30),
-				Secure:  true,
-			})
-			session.Values[loginKey] = login
-			if err := session.Save(r, w); err != nil {
-				serverError("Save oauth session", err)
-				return
-			}
+			if user != nil {
+				login := user.Login
+				data.Login = true
+				// Saves login. We save the login under 2 cookies. One for the use of client to render the
+				// data and one encoded for server to verify the identity of the authenticated user.
+				http.SetCookie(w, &http.Cookie{
+					Name:    loginSession,
+					Value:   login,
+					Path:    "/",
+					Expires: time.Now().Add(time.Hour * 24 * 30),
+					Secure:  true,
+				})
+				session.Values[loginKey] = login
+				if err := session.Save(r, w); err != nil {
+					serverError("Save oauth session", err)
+					return
+				}
 
-			// Construct query
-			ghc := da.github.GitHubClientWithAccessToken(token.AccessToken) // TODO(fejta): we should not recreate the client
```

Comment:
Please also remove the now unused `github` field from the `DashboardAgent` struct